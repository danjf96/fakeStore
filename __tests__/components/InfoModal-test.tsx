/**
* @format
*/

import 'react-native';
import React from 'react';
import InfoModal from '../../src/components/infoModal';

// Note: test renderer must be required after react-native.
import { fireEvent, render } from 'react-native-testing-library';

describe('renders correctly', () => {
    const pressSuccess = jest.fn()
    const pressCancel = jest.fn()

    const component = <InfoModal 
        title='TITLE'
        text='TEXT'
        textSuccess='OK'
        textCancel='CANCEL'
        btnCancel={pressSuccess}
        btnSuccess={pressCancel}
        visible={true}
    />

    it('basic', () => {
        const screen = render(component);
        expect(screen.toJSON()).toMatchSnapshot()
    });



    it('text', () => {
        const { getAllByText } = render(component);
        expect(getAllByText(/TITLE|TEXT|OK|CANCEl/)).toBeTruthy()
    });

    // it('functions', () => {
    //     const { getByText } = render(component);

    //     fireEvent(getByText('OK'), 'onPress')
    //     expect(pressSuccess).toBeCalled()

    //     fireEvent(getByText('CANCEL'), 'onPress')
    //     expect(pressCancel).toBeCalled()

    // });
})

