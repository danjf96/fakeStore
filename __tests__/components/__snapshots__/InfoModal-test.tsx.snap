// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly basic 1`] = `
<Modal
  hardwareAccelerated={false}
  style={
    Object {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#696969",
          "height": "100%",
          "opacity": 0.3,
          "position": "absolute",
          "width": "100%",
          "zIndex": 0,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "flexDirection": "column",
          "opacity": 1,
          "paddingVertical": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#696969",
            "fontFamily": "WorkSans-Regular",
            "fontSize": 20,
            "fontWeight": "700",
            "marginBottom": 8,
            "textAlign": "center",
          }
        }
      >
        TITLE
      </Text>
      <Text
        style={
          Object {
            "color": "#B5B5B5",
            "fontFamily": "WorkSans-Regular",
            "fontSize": 16,
            "fontWeight": "600",
            "marginBottom": 20,
            "textAlign": "center",
            "width": "90%",
          }
        }
      >
        TEXT
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#E7E7E7",
            "borderRadius": 0,
            "borderTopWidth": 1,
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "WorkSans-Regular",
                "fontSize": 14,
                "fontWeight": "700",
              },
              Object {
                "color": "#5A9ACF",
              },
            ]
          }
        >
          OK
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#E7E7E7",
            "borderRadius": 0,
            "borderTopWidth": 1,
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "WorkSans-Regular",
                "fontSize": 14,
                "fontWeight": "700",
              },
              Object {
                "color": "#FD746A",
              },
            ]
          }
        >
          CANCEL
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
